backend/src/judge/middleware/cpp.ts

```diff
@@ -66,33 +70,29 @@ export class CppMiddleware extends JudgeMiddleware {
       }
     }
 
-    const joinedVariableNames = codePrototype.arguments
-      .map((arg) => arg.name)
-      .join(",");
-
-    let printOutput;
+    let expectedOutput = "";
+    let isEqual = "";
     if (codePrototype.returnType.includes("vector")) {
-      printOutput =
-        `  std::stringstream ss;\n` +
-        `  ss << "[";\n` +
-        `  for (size_t i = 0; i < res.size(); i++) {\n` +
-        `    if (i != 0) {\n` +
-        `      ss << ", ";\n` +
-        `    }\n` +
-        `    ss << res[i];\n` +
-        `  }\n` +
-        `  ss << "]";\n` +
-        `  std::cout << ss.str() << std::endl;\n`;
+      const output = this.output.replace(/^\[/, "{").replace(/\]$/, "}");
+      expectedOutput = `${codePrototype.returnType} expectedOutput${output};`;
+      isEqual = `bool isEqual = std::equal(res.begin(), res.end(), expectedOutput.begin());`;
```

Comment:
Do we need to check that `res` is the same size as `expectedOutput`?

Reply:
Yep I forgot about this. Thanks!